src/tools/dev/xml/Field.h

```diff
@@ -1348,7 +1351,8 @@ class Att : public virtual Field
     {
         attType = t;
         AttType = t;
-        AttType[0] = AttType[0].toUpper();
+        if (!t.isEmpty())
+            AttType[0] = AttType[0].toUpper();
```

Comment:
Are there any issues down the line that could come up if the string is empty? This fix addresses the "first" place where a segfault could occur, but now that this is fixed, do we handle this case correctly otherwise?

Reply:
Nothing with the tool or generating code fails. But without subtype being filled in, things won't work correctly.
I didn't add other types of error checking as there was none in the first place. Behavior of QString changed subtly between versions which is why this segv wasn't caught before.

Reply:
Do you think we should open a new ticket to address the missing subtype case? Or is it sufficient to leave as is? My head isn't fully wrapped around this at the moment.

Reply:
In this first case, the constructor, it is okay that the 't' (representing subtype) is empty. This occurs for a new or existing field whose type is being changed to 'att' or 'attvector'. In which case the 'subtype' widget is enabled by the action of choosing 'att' or 'attvector' but is by default empty until user takes action.

In the second case, 'SetSubtype', it would make sense to have error handling for empty string.  Perhaps, have the 'subtype' widget in xmledit do the error handling before 'SetSubtye' is called on the Field. Will take a look.

Reply:
Okay, I added error checking for empty 'subtype' when file is being written (XMLDocument.C). This prevents bad .xml files and bad code-gen'd files.
Thanks for the input!

Reply:
Thanks for investigating! 💪